src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -496,6 +496,8 @@ private static ScriptType ParseScriptType(string scriptFilePath)
 
         internal Collection<FunctionDescriptor> ReadFunctions(List<FunctionMetadata> metadatas, IEnumerable<FunctionDescriptorProvider> descriptorProviders)
         {
+            metadatas = ApplyMaximumFunctionCount(metadatas);
```

Comment:
It's your call, we can stay with what you have. The whole thing is an edge case, so the unnecessary processing is hardly a concern.